### ISSUE TITLE
ValueError: Unknown optimizer: Optimizer

### ISSUE THREAD

**Issue by Arpitharavi08** [run-ai/runai#11]
Hello team,

    I really like this option but when i try to run my model on the test data i get this error. Could you help me with it.

I also tried compile = False while loading model and also the custom objects option.

Any solution will be appreciated.

**Comment by noa-neria** [comment]
Hi,
The error message suggests that you pass the optimizer as a string.
However, the model should be compiled with an instance of runai wrapped optimizer.
For example:

import keras
import runai.ga

STEPS = 5
my_optimizer = keras.optimizers.SGD()
ga_optimizer = runai.ga.keras.optimizers.Optimizer(my_optimizer, STEPS)

model = keras.models.Sequential(...)
model.compile(optimizer=ga_optimizer,...)

**Comment by ArpithaFAU** [comment]
Thank you very much i will try this

**Comment by Arpitharavi08** [comment]
@noa-neria  i have tried your suggestion and i still have the same error when i try to use my model for testing.
Error is as follows:

 line 140, in deserialize_keras_object
    ': ' + class_name)
ValueError: Unknown optimizer: Optimizer

Could there be any other way to do this?

**Comment by noa-neria** [comment]
deserialize_keras_object is responsible for creation of a keras object from a name (string) or configuration dictionary, so it seems that somewhere in the code the optimizer is not passed as keras object.
Can you please share your code?

**Comment by Arpitharavi08** [comment]
vgg = vgg16.VGG16(weights="imagenet", input_shape= in_shape, include_top = False)

    for layer in vgg.layers[0:-1]:
        layer.trainable = True
    outputs = vgg.output
    vgg.summary()

    return keras.Model(vgg.input, outputs)
vggmodel = build_vgg((HEIGHT,WIDTH,channel))

input_layer = keras.Input(shape=(timestep,HEIGHT,WIDTH,channel))
td = keras.layers.wrappers.TimeDistributed(vggmodel)(input_layer)
print(np.shape(td))
com=np.asarray(td.get_shape().as_list()[2:]).prod()
im_layer = Reshape(target_shape=(timestep, com))(td)
x = LSTM(128, return_sequences= True,kernel_initializer='glorot_uniform', kernel_regularizer = regularizers.l2(0.001), dropout=0.5)(im_layer)
fc1 = Dense(128, activation='relu')(x)
dp_1 = Dropout(0.5)(fc1)
fl1 = Flatten()(dp_1)
print(np.shape(fl1))
final_output = Dense(num_classes, activation='sigmoid')(fl1)
model = Model(input_layer,final_output)
STEPS = 6
my_optimizer = keras.optimizers.Adam()
ga_optimizer = runai.ga.keras.optimizers.Optimizer(my_optimizer, STEPS)

model.compile(loss='binary_crossentropy',
              optimizer= ga_optimizer,
              metrics=['accuracy'])


This is my model

**Comment by noa-neria** [comment]
The imports are not shown above.
It's essential to do "import keras" and not tensorflow.keras

I compiled your model  with the right import.

**Comment by Arpitharavi08** [comment]
Yes i am using import keras. I can compile and start the training but when i want to reuse the model i get this error.

**Comment by noa-neria** [comment]
OK. How do you reuse the model? Is there any serialization in between?

**Comment by noa-neria** [comment]
Loading with compile=False, and then compiling the model with the ga_optimizer instance should work.

**Comment by Arpitharavi08** [comment]
> 
> 
> OK. How do you reuse the model? Is there any serialization in between?

There is no serialization, i just load the weights and test it using the predict method.

**Comment by Arpitharavi08** [comment]
> 
> 
> Loading with compile=False, and then compiling the model with the ga_optimizer instance should work.

I have tried this before and i got the same error. However, i can try it again.
Have you tested this method before and has it worked for you?? Please let me know

**Comment by noa-neria** [comment]
I tested with the mlp example and it worked:

model.save('test.model')
model = keras.models.load_model('test.model', compile=False)
optimizer = runai.ga.keras.optimizers.Adam(STEPS)
model.compile(loss='categorical_crossentropy',
                           optimizer=optimizer,
                           metrics=['accuracy'])
score = model.evaluate(x_test, y_test, verbose=0)

**Comment by Arpitharavi08** [comment]
> I tested with the mlp example and it worked:
> 
> model.save('test.model')
> model = keras.models.load_model('test.model', compile=False)
> optimizer = runai.ga.keras.optimizers.Adam(STEPS)
> model.compile(loss='categorical_crossentropy',
> optimizer=optimizer,
> metrics=['accuracy'])
> score = model.evaluate(x_test, y_test, verbose=0)

This worked now. Thank you so much.
